tests/API1/testjsonserialization.py

```diff
@@ -0,0 +1,125 @@
+    f = param.Date(default=datetime.datetime.now())
+    g = param.Tuple(default=(1,2,3), length=3)
+    h = param.NumericTuple(default=(1,2,3,4))
+    i = param.XYCoordinates(default=(32.1, 51.5))
+    j = param.Integer(default=1)
+    k = param.Range(default=(1.1,2.3), bounds=(1,3))
+    l = param.String(default='baz', allow_None=True)
+    m = param.ObjectSelector(default=3, objects=[3,'foo'], allow_None=False)
+    n = param.ListSelector(default=[1,4,5], objects=[1,2,3,4,5,6])
+    o = param.DataFrame(default=pd.DataFrame({'A':[1.,2.,3.], 'B':[1.1,2.2,3.3]}), columns=2)
+    p = param.DataFrame(default=pd.DataFrame({'A':[1.,2.,3.], 'B':[1.1,2.2,3.3]}), columns=(1,4), rows=(2,5))
+    q = param.DataFrame(default=pd.DataFrame({'A':[1.1,2.2,3.3],
+                                              'B':[1.1,2.2,3.3]}), columns=['A', 'B'])
+
+test = TestSet(a=29)
+
+
+class TestJSONSerialization(API1TestCase):
+
+    def test_serialize_integer_class(self):
+        serialization = TestSet.param.serialize_parameters(subset=['a'], mode='json')
+        deserialized = json.loads(serialization)
+        self.assertEqual(TestSet.a, 5)
+        self.assertEqual(deserialized, {'a':TestSet.a})
+
+    def test_serialize_integer_instance(self):
+        serialization = test.param.serialize_parameters(subset=['a'], mode='json')
+        deserialized = json.loads(serialization)
+        self.assertEqual(test.a, 29)
+        self.assertEqual(deserialized, {'a':test.a})
+
+    def test_serialize_integer_schema_class(self):
+        if validate is None:
+            raise SkipTest('jsonschema needed for schema validation testing')
+        param_schema = TestSet.param.schema(safe=True, subset=['a'], mode='json')
+        schema = {"type" : "object", "properties" : param_schema}
+        serialized = json.loads(TestSet.param.serialize_parameters(subset=['a']))
+        self.assertEqual({'a':
+                          {'type': 'integer', 'minimum': 2, 'exclusiveMaximum': 30,
+                           'description': 'Example doc', 'title': 'A'}},
+                         param_schema)
+        validate(instance=serialized, schema=schema)
+
+    def test_serialize_integer_schema_class_invalid(self):
+        if validate is None:
+            raise SkipTest('jsonschema needed for schema validation testing')
+        param_schema = TestSet.param.schema(safe=True, subset=['a'], mode='json')
+        schema = {"type" : "object", "properties" : param_schema}
+        serialized = json.loads(TestSet.param.serialize_parameters(subset=['a']))
+        self.assertEqual({'a':
+                          {'type': 'integer', 'minimum': 2, 'exclusiveMaximum': 30,
+                           'description': 'Example doc', 'title': 'A'}},
+                         param_schema)
+
+        exception = "1 is not of type 'object'"
+        with self.assertRaisesRegexp(ValidationError, exception):
+            validate(instance=1, schema=schema)
+
+
```

Comment:
Nitpicky (and I know this isn't quite ready) but can we standardize on one newline between methods like the rest of param?

Reply:
Addressed in one of the following commits.